thz/ThzProtocol.py

```diff
@@ -0,0 +1,1340 @@
+#########################################################################
+#  Ported by toggle, 2014-2016
+#########################################################################
+#  This software is based on the FHEM implementation
+#  https://github.com/mhop/fhem-mirror/blob/master/fhem/FHEM/00_THZ.pm
+#
+#  THZ plugin for SmartHome.py.   http://mknx.github.com/smarthome/
+#
+#  This plugin is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This plugin is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this plugin. If not, see <http://www.gnu.org/licenses/>.
+#########################################################################
+
+# This file implements message definitions and
+# protocol encoding and decoding function for THZ/LWZ.
+
+import time
+import sys
+import logging
+import threading
+import datetime
+import struct
+from . import PortHandler
+
+# mode mappings to plain text
+OpMode = {'00': 'emergency', '01': 'standby', '03': 'DAYmode', '04': 'setback',
+          '05': 'DHWmode', '11': 'automatic', '14': 'manual' }
+
+OpModeHC = {'01': 'normal', '02': 'setback', '03': 'standby', '04': 'restart',
+          '05': 'restart' }
+
+logger = logging.getLogger('THZ')
+
+####################################################################
+#
+# Decoding functions
+#
+####################################################################
```

Comment:
Maybe you can move all this functions into a class `DecodingUtil` and movin the logger into this class instead of having them globally (functions and logger)?